controllers/front/ProductController.php

```diff
@@ -944,49 +944,89 @@ public function getCategory()
+    private function getIdProductAttributeByRequestOrGroup()
     {
         $requestedIdProductAttribute = (int) Tools::getValue('id_product_attribute');
 
-        if ($useGroups === true) {
-            $groups = Tools::getValue('group');
+        $groupIdProductAttribute = $this->getIdProductAttributeByGroup();
+        $requestedIdProductAttribute = null !== $groupIdProductAttribute ? $groupIdProductAttribute : $requestedIdProductAttribute;
 
-            if (!empty($groups)) {
-                $requestedIdProductAttribute = (int) Product::getIdProductAttributeByIdAttributes(
-                    $this->product->id,
-                    $groups,
-                    true
-                );
-            }
-        }
+        return $this->tryToGetAvailableIdProductAttribute($requestedIdProductAttribute);
+    }
 
+    /**
+     * If the PS_DISP_UNAVAILABLE_ATTR functionality is enabled, this method check
+     * if $checkedIdProductAttribute is available.
+     * If not try to return the first available attribute, if none are available
+     * simply returns the input
+     *
+     * @param int $checkedIdProductAttribute
+     * @return int
+     */
+    private function tryToGetAvailableIdProductAttribute($checkedIdProductAttribute)
+    {
         if (!Configuration::get('PS_DISP_UNAVAILABLE_ATTR')) {
-            $productAttributes = array_filter(
+            $availableProductAttributes = array_filter(
                 $this->product->getAttributeCombinations(),
                 function ($elem) {
                     return $elem['quantity'] > 0;
                 }
             );
-            $productAttribute = array_filter(
-                $productAttributes,
-                function ($elem) use ($requestedIdProductAttribute) {
-                    return $elem['id_product_attribute'] == $requestedIdProductAttribute;
+
+            $availableProductAttribute = array_filter(
+                $availableProductAttributes,
+                function ($elem) use ($checkedIdProductAttribute) {
+                    return $elem['id_product_attribute'] == $checkedIdProductAttribute;
                 }
             );
 
-            if (empty($productAttribute) && !empty($productAttributes)) {
-                return (int)array_shift($productAttributes)['id_product_attribute'];
+            if (empty($availableProductAttribute) && count($availableProductAttributes)) {
+                return (int)array_shift($availableProductAttributes)['id_product_attribute'];
             }
         }
 
-        return $requestedIdProductAttribute;
+        return $checkedIdProductAttribute;
+    }
+
+    /**
+     * Return id_product_attribute by the group request parameter.
+     *
+     * @return int|null
+     * @throws PrestaShopException
+     */
+    private function getIdProductAttributeByGroup()
+    {
+        $idProductAttribute = null;
+
+        $groups = Tools::getValue('group');
+        if (!empty($groups)) {
```

Comment:
No need variable here

```php
if (!empty($groups)) {
  return null;
}

return Product::getIdProductAttributeByIdAttributes(
    $this->product->id,
    $groups,
     true
);
```

Reply:
rather
`if (empty($groups)) {`
but ok I change it

Reply:
Yes, wrong copy & paste :P

Reply:
done!